File: source/scripts/python/ducktype/source/ducktype.py
#!/usr/bin/python3.5

def multiply(left, right):
	result = left * right;
	print(left, ' * ', right, ' = ', result);
	return result;

def divide(left, right):
	if right != 0.0:
		result = left / right;
		print(left, ' / ', right, ' = ', result);
	else:
		print('Invalid right operand: ', right);
	return result;

def sum(left, right):
	result = left + right;
	print(left, ' + ', right, ' = ', result);
	return result;

def hello():
	print('Hello World from Python!!');
	return;

def strcat(left, right):
	result = left + right;
	print(left, ' + ', right, ' = ', result);
	return result;

def old_style(left: int, right: int) -> int:
	result = left + right;
	print(left, ' + ', right, ' = ', result);
	return result;

def mixed_style(left, right: int) -> int:
	result = left + right;
	print(left, ' + ', right, ' = ', result);
	return result;

def mixed_style_noreturn(left, right: int):
	result = left + right;
	print(left, ' + ', right, ' = ', result);
	return result;

File: source/scripts/ruby/ducktype/source/ducktype.rb
#!/usr/bin/ruby

def say_hello(value)
	result = 'Hello ' + value + '!'
	puts(result)
	return result
end

def say_multiply(left, right)
	result = left * right
	puts('Multiply', result, '!')
	return result
end

def say_null()
	puts('Helloooo from null method!')
end

def get_second(first, second)
	puts('Second value is', second)
	return second
end

def mixed(a, b, c: Fixnum, d: Fixnum)
	puts('Mixed arguments:', a, b, c, d)
	return a + b + c + d
end

def map_style(a: Fixnum, b: Fixnum)
	puts('Old style arguments:', a, b)
	return a + b
end
